autogpt_platform/backend/backend/server/v2/store/db.py

```diff
@@ -765,28 +764,20 @@ async def get_agent(
         )
 
         if not store_listing_version or not store_listing_version.Agent:
-            raise fastapi.HTTPException(
-                status_code=404,
-                detail=f"Store listing version {store_listing_version_id} not found",
-            )
+            logger.error(f"Store listing version {store_listing_version_id} not found")
+            return None
 
         graph_id = store_listing_version.agentId
         graph_version = store_listing_version.agentVersion
         graph = await backend.data.graph.get_graph(graph_id, graph_version)
 
         if not graph:
-            raise fastapi.HTTPException(
-                status_code=404,
-                detail=(
-                    f"Agent #{graph_id} not found "
-                    f"for store listing version #{store_listing_version_id}"
-                ),
-            )
+            logger.error(f"Agent {agent.id} not found")
```

Comment:
If you leave this check it should be a `raise DatabaseError`, since a `StoreListingVersion` can't not have an associated `AgentGraph` afaik